### PR TITLE
Testing methods using rspec

### PR DESCRIPTION
**Pull request by carloso0114**
In this case we work using rspec testing each methods checking if the inputs are good or wrong. 
Most of the test cases are valid, and some of them are asking for false cases that could possibly happen.

### CODE REVIEW

File: spec/tests_spec.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    end
+    it 'return true if a row/column/diagonal has equal values' do
+      new_game
+      new_game.reset_board
+      new_game.update_board([0, 2], 'O')
+      new_game.update_board([1, 1], 'O')
+      new_game.update_board([2, 0], 'O')
+      expect(new_game.win?).to be_truthy
+    end
+  end
```
Comment (MarilenaRoque) [inline spec/tests_spec.rb]: These tests were made in the first 'describe' block. Kindly remove the repeated tests.

File: spec/tests_spec.rb [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      it 'return true if a diagonal has equal values' do
+        new_game
+        new_game.reset_board
+        new_game.update_board([0, 2], 'O')
+        new_game.update_board([1, 1], 'O')
+        new_game.update_board([2, 0], 'O')
+        expect(new_game.win?).to be_truthy
+      end
+    end
+  end
```
Comment (MarilenaRoque) [inline spec/tests_spec.rb]: Please, test negative scenarios for the #win? method.

**Review by MarilenaRoque — CHANGES_REQUESTED** [review]
# Changes Required ♻️ 

## Hello! :rocket:

### Great job on this milestone, so far! 👏🏻 

- The separation of concerns in your methods made really easy to understand and review the tests. Congratulations on that! 🥳 

### Although, we need to make some changes to achieve all the requirements.

- [x] Delete or replace the duplicated tests for the '#win?' method.

- [x] Add more tests for negative scenarios.

Please check more details in my comments below.

Feel free to message me on Slack if you have questions.
Happy Coding! 😃 
**Mari Roque**

**Review by javitocor — APPROVED** [review]
# STATUS APPROVED 🥇 7

Congratulations team! 🥳 
You have met all the requirements and implemented the previous TSE suggestions! 💯 
Good job! 👍 
Remember to merge it, fill up the completion form 📑 and you can move into the next project 🚀 
Good luck with it! 😺
